source/ota_mqtt.c

```diff
@@ -362,14 +491,25 @@ static OtaMqttStatus_t unsubscribeFromJobNotificationTopic( const OtaAgentContex
     char pJobTopic[ TOPIC_GET_NEXT_ACCEPTED_BUFFER_SIZE ];
     uint16_t topicLen = 0;
 
+    /* NULL-terminated list of topic string parts */
+    const char * topicStringParts[] =
+    {
+        MQTT_API_THINGS,
+        NULL, /* Thing Name not available at compile time, initialized below */
+        /* pAgentCtx->pThingName, */
+        MQTT_API_JOBS_NOTIFY_NEXT,
+        NULL
+    };
+
     assert( pAgentCtx != NULL );
 
     /* Try to unsubscribe from the first of two job topics. */
-    /* coverity[misra_c_2012_rule_21_6_violation] */
-    topicLen = ( uint16_t ) snprintf( pJobTopic,
-                                      sizeof( pJobTopic ),
-                                      pOtaJobsNotifyNextTopicTemplate,
-                                      pAgentCtx->pThingName );
+
+    topicStringParts[ 1 ] = ( const char * ) pAgentCtx->pThingName;
```

Comment:
Could we use an enum instead of using the index directly? It may make which part of the string is being set. I think this is already kind of obvious due to what is being assigned though, so this is optional.

Reply:
Let me look into it – it may not make the code any clearer to read, since every topic & payload are slightly different.  I might use `const int` values in the functions themselves, that might be easy to read and more maintainable.